From a1729a003f820289c403360d43f10b7f84c5a03c Mon Sep 17 00:00:00 2001
From: Janne Juntunen <janne.juntunen@qt.io>
Date: Tue, 2 Aug 2022 11:03:08 +0300
Subject: [PATCH] Fix qt toolchain for SDK use

Set initial host path based on the location of the
QtConfigDependencies.cmake.in file. This helps us avoid need to relocate
the file in SDK.

The calculated paths for QT_TOOLCHAIN_RELOCATABLE paths point to
host sysroot which must not be used when cross-compiling other projects.

Change-Id: I8d23f76e6e7abb2651b414661304ad58e5edd522
---
 cmake/QtConfigDependencies.cmake.in | 6 ++++--
 cmake/qt.toolchain.cmake.in         | 4 ++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/cmake/QtConfigDependencies.cmake.in b/cmake/QtConfigDependencies.cmake.in
index c17ae28bac..4cc804c27a 100644
--- a/cmake/QtConfigDependencies.cmake.in
+++ b/cmake/QtConfigDependencies.cmake.in
@@ -1,8 +1,10 @@
 set(@INSTALL_CMAKE_NAMESPACE@_FOUND FALSE)
 
 set(__qt_platform_requires_host_info_package "@platform_requires_host_info_package@")
-set(__qt_platform_initial_qt_host_path "@qt_host_path_absolute@")
-set(__qt_platform_initial_qt_host_path_cmake_dir "@qt_host_path_cmake_dir_absolute@")
+get_filename_component(__qt_toolchain_initial_qt_host_path
+    "${CMAKE_CURRENT_LIST_DIR}/.." ABSOLUTE)
+get_filename_component(__qt_toolchain_initial_qt_host_path_cmake_dir
+    "${CMAKE_CURRENT_LIST_DIR}/../lib/cmake" ABSOLUTE)
 
 _qt_internal_setup_qt_host_path(
     "${__qt_platform_requires_host_info_package}"
diff --git a/cmake/qt.toolchain.cmake.in b/cmake/qt.toolchain.cmake.in
index 15cf7a432e..03de9288df 100644
--- a/cmake/qt.toolchain.cmake.in
+++ b/cmake/qt.toolchain.cmake.in
@@ -67,8 +67,8 @@ get_filename_component(QT_TOOLCHAIN_RELOCATABLE_CMAKE_DIR "${CMAKE_CURRENT_LIST_
 # Instead of collapsing the search prefix (which is the case when one is a subdir of the other),
 # it concatenates them creating an invalid path. Workaround it by setting the root path to the
 # Qt install prefix, and the prefix path to the lib/cmake subdir.
-list(PREPEND CMAKE_PREFIX_PATH "${QT_TOOLCHAIN_RELOCATABLE_CMAKE_DIR}")
-list(PREPEND CMAKE_FIND_ROOT_PATH "${QT_TOOLCHAIN_RELOCATABLE_INSTALL_PREFIX}")
+#list(PREPEND CMAKE_PREFIX_PATH "${QT_TOOLCHAIN_RELOCATABLE_CMAKE_DIR}")
+#list(PREPEND CMAKE_FIND_ROOT_PATH "${QT_TOOLCHAIN_RELOCATABLE_INSTALL_PREFIX}")
 
 # Let CMake load our custom platform modules.
 # CMake-provided platform modules take precedence.
